yarn-project/end-to-end/src/fixtures/utils.ts

```diff
@@ -490,3 +505,25 @@ export async function expectMapping<K, V>(
 
   expect(outputs).toEqual(expectedOutputs);
 }
+
+/**
+ * Deploy the protocol contracts to a running instance.
+ */
+export async function deployPublicProtocolContracts(deployer: Wallet) {
+  // "deploy" the Gas token as it contains public functions
+  const canonicalGasToken = getCanonicalGasToken();
+
+  if (await deployer.isContractClassPubliclyRegistered(canonicalGasToken.contractClass.id)) {
+    return;
```

Comment:
Is this assuming that if it is publicly registered then the instance will also have been deployed?